caddyhttp/basicauth/setup_test.go

```diff
@@ -70,6 +70,30 @@ md5:$apr1$l42y8rex$pOA2VJ0x/0TwaFeAF9nX61`
 		}`, false, "pwd", []Rule{
 			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
 		}},
+		{`basicauth user pwd {
+			path /resource1
+			path /resource2
+		}`, false, "pwd", []Rule{
+			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
+		}},
+		{`basicauth user pwd {
+			/resource1
+			realm "Secure resources"
+			path /resource2
```

Comment:
Good point, about the optional path. I forgot about that.

What if we keep your changes as-is, but drop the "path" keyword? In other words, each line is a list of resources, unless the line starts with "realm" (and has two arguments). That way, there's only one way to do it in that case.